libhb/encavcodec.c

```diff
@@ -1572,6 +1587,8 @@ const char* const* hb_av_preset_get_names(int encoder)
         case HB_VCODEC_FFMPEG_NVENC_H264:
         case HB_VCODEC_FFMPEG_NVENC_H265:
         case HB_VCODEC_FFMPEG_NVENC_H265_10BIT:
+        case HB_VCODEC_FFMPEG_NVENC_AV1: //TODO: preset names
```

Comment:
Are the preset names the same for AV1?

Reply:
The preset name are the same for all NVENC codecs (also in FFmpeg). Thank you for reminding me of that. When AV1 support was added for OBS, they migrated the users automatically to the new presets https://github.com/obsproject/obs-studio/blob/f55b251fa8da4bc1304e48293b947f02998ad038/plugins/obs-ffmpeg/jim-nvenc.c#L467-L543, because  AV1 encode API rejected the deprecated old presets (deprecated since Video SDK 10).

The new presets have been around for a while https://developer.nvidia.com/blog/introducing-video-codec-sdk-10-presets/. https://docs.nvidia.com/video-technologies/video-codec-sdk/nvenc-preset-migration-guide/index.html provides a migration guide. This is how preset names are mapped on the FFmpeg side: https://github.com/FFmpeg/FFmpeg/blob/a3bf63c26e6197a9b6e490538667368530a9dcd3/libavcodec/nvenc.c#L178-L212 so you FFmpeg will already map old names to new presets. But maybe at some time Handbrake could expose the new p1-p7 presets to users as OBS does at some point.
![image](https://user-images.githubusercontent.com/7189118/224482476-12599e07-c13b-49bc-b360-f9364ec990f8.png)

See https://github.com/FFmpeg/FFmpeg/blob/58d6426870ed10e5eeed98418dc3b7b2d4efe2c6/libavcodec/nvenc_h264.c#L28-L65 (H264) vs https://github.com/FFmpeg/FFmpeg/blob/58d6426870ed10e5eeed98418dc3b7b2d4efe2c6/libavcodec/nvenc_av1.c#L29-L45 (AV1) for available presets names in FFmpeg. Strange that I can't find (`fastest`/`slowest` there. Did got removed?)

Reply:
I also missed the AV1 profile names :disappointed:

Reply:
My fault. Handbrake already uses P1-P7 https://github.com/theHamsta/HandBrake/blob/17c680d5f9f7f5b0f4f13305e887999f059b04eb/libhb/nvenc_common.c#L93-L117